superset-frontend/src/components/Chart/chartAction.js

```diff
@@ -269,9 +269,13 @@ export function runAnnotationQuery({
       return Promise.resolve();
     }
 
-    const granularity = fd.time_grain_sqla || fd.granularity;
-    fd.time_grain_sqla = granularity;
-    fd.granularity = granularity;
+    // remap from controls to formData naming; in the original formData the
+    // `granularity` attribute represents the time grain, but in the request
+    // payload it corresponds to the name of the column where the time grain
+    // should be applied...
```

Comment:
it will be helpful to add sample data for time_grain and granularity as inline comment. .e.g
original `granularity` can be `P1D` but in request payload expect to be column name like `Date`